pkg/ipam/crd.go

```diff
@@ -515,8 +525,40 @@ func (a *crdAllocator) buildAllocationResult(ip net.IP, ipInfo *ipamTypes.Alloca
 
 		result = nil
 		err = fmt.Errorf("unable to find ENI %s", ipInfo.Resource)
+
+	// In AlibabaCloud mode,
+	case ipamOption.IPAMAlibabaCloud:
+		for _, eni := range a.store.ownNode.Status.AlibabaCloud.ENIs {
+			if eni.NetworkInterfaceID != ipInfo.Resource {
+				continue
+			}
+			result.PrimaryMAC = eni.MacAddress
+			result.CIDRs = []string{eni.VSwitch.CidrBlock}
+
+			gw, err2 := metadata.GetENIGateway(context.TODO(), eni.MacAddress)
+			if err2 != nil {
+				err = fmt.Errorf("unable get gw from ENI %w", err2)
+				return
+			}
+			result.GatewayIP = gw.IP.String()
+
+			links, err2 := netlink.LinkList()
+			if err2 != nil {
+				err = fmt.Errorf("unable get link %w", err2)
+				return
+			}
+			for _, link := range links {
+				if link.Attrs().HardwareAddr.String() == eni.MacAddress {
+					result.InterfaceNumber = strconv.Itoa(link.Attrs().Index)
+					break
+				}
+			}
+			return
+		}
 	}
 
+	result = nil
+	err = fmt.Errorf("unable to find ENI %s", ipInfo.Resource)
```

Comment:
Looking at the other modes, it looks like the two lines above should be _inside_ the `case ipamOption.IPAMAlibabaCloud:`, not _below the switch-case_? Or did you mean to refactor this and to have this error message just once for the three modes, in which case we should remove the corresponding lines for AWS/Azure? I think we'll have the error message twice for AWS/Azure if the ENI is not found otherwise.